packages/oncokb-frontend-commons/src/components/RefComponent.tsx

```diff
@@ -11,46 +11,75 @@ export default class RefComponent extends React.Component<{
     componentType: 'tooltip' | 'linkout';
 }> {
     render() {
-        const parts = this.props.content.split(/pmid|nct/i);
+        const parts = this.props.content.split(/pmid|nct|abstract/i);
 
         if (parts.length < 2) {
             return <span>{this.props.content}</span>;
         }
 
-        const ids = parts[1].match(/[0-9]+/g);
-
-        if (!ids) {
-            return <span>{this.props.content}</span>;
-        }
+        // Slicing from index 1 to end to rejoin the abstract title and link
+        // when there's the case that they also contain the string 'Abstract'
+        // Example :
+        //     (Abstract: Fakih et al. Abstract# 3003, ASCO 2019. https://meetinglibrary.asco.org/record/12411/Abstract)
+        const abstractParts = parts
+            .slice(1)
+            .join('Abstract')
+            .split(/(?=http)/i);
+        const isAbstract = !(abstractParts.length < 2);
 
+        let abstract = '';
+        let abstractLink = '';
+        const ids = parts[1].match(/[0-9]+/g);
         let prefix: string | undefined;
-
-        if (this.props.content.toLowerCase().indexOf('pmid') >= 0) {
-            prefix = 'PMID: ';
-        } else if (this.props.content.toLowerCase().indexOf('nct') >= 0) {
-            prefix = 'NCT';
-        }
-
         let link: JSX.Element | undefined;
 
-        if (prefix) {
+        if (isAbstract) {
+            abstract = abstractParts[0].replace(/^[:\s]*/g, '').trim();
+            abstractLink = abstractParts[1].replace(/[\\)]*$/g, '').trim();
+            prefix = 'Abstract: ';
             link = (
                 <a
                     target="_blank"
-                    href={getNCBIlink(`/pubmed/${ids.join(',')}`)}
+                    rel="noopener noreferrer"
+                    href={abstractLink}
                 >
-                    {`${prefix}${ids.join(',')}`}
+                    {`${prefix}${abstract}`}
                 </a>
             );
+        } else {
+            if (!ids) {
+                return <span>{this.props.content}</span>;
+            }
+
+            if (this.props.content.toLowerCase().includes('pmid')) {
+                prefix = 'PMID: ';
+            } else if (this.props.content.toLowerCase().includes('nct')) {
+                prefix = 'NCT';
+            }
+
+            if (prefix) {
+                link = (
+                    <a
+                        target="_blank"
+                        rel="noopener noreferrer"
+                        href={getNCBIlink(`/pubmed/${ids.join(',')}`)}
+                    >
+                        {`${prefix}${ids.join(', ')}`}
+                    </a>
+                );
```

Comment:
This is partially duplicate code now since we have a similar code segment in the `if (isAbstract)` block.

Reply:
Made some changes to move the code outside of the `if (isAbstract) ... else ...`.